experiments/ClimaEarth/run_amip.jl

```diff
@@ -97,640 +97,654 @@ include("user_io/arg_parsing.jl")
-sypd = CA.simulated_years_per_day(es)
-n_atmos_steps = atmos_sim.integrator.step
-walltime_per_atmos_step = es.walltime / n_atmos_steps
-@info "SYPD: $sypd"
-@info "Walltime per Atmos step: $(walltime_per_atmos_step)"
-
-## Save the SYPD and allocation information
-if ClimaComms.iamroot(comms_ctx)
-    open(joinpath(dir_paths.artifacts, "sypd.txt"), "w") do sypd_filename
-        println(sypd_filename, "$sypd")
+    This is where the full coupling loop, `solve_coupler!` is called for the full timespan of the simulation.
+    We use the `ClimaComms.@elapsed` macro to time the simulation on both CPU and GPU, and use this
+    value to calculate the simulated years per day (SYPD) of the simulation.
+    =#
+    walltime = ClimaComms.@elapsed comms_ctx.device begin
+        s = CA.@timed_str begin
+            solve_coupler!(cs)
+        end
     end
+    @info(walltime)
+
+    ## Use ClimaAtmos calculation to show the simulated years per day of the simulation (SYPD)
+    es = CA.EfficiencyStats(tspan, walltime)
+    sypd = CA.simulated_years_per_day(es)
+    n_atmos_steps = atmos_sim.integrator.step
+    walltime_per_atmos_step = es.walltime / n_atmos_steps
+    @info "SYPD: $sypd"
+    @info "Walltime per Atmos step: $(walltime_per_atmos_step)"
+
+    ## Save the SYPD and allocation information
+    if ClimaComms.iamroot(comms_ctx)
+        open(joinpath(dir_paths.artifacts, "sypd.txt"), "w") do sypd_filename
+            println(sypd_filename, "$sypd")
+        end
+
+        open(joinpath(dir_paths.artifacts, "walltime_per_atmos_step.txt"), "w") do walltime_per_atmos_step_filename
+            println(walltime_per_atmos_step_filename, "$(walltime_per_atmos_step)")
+        end
 
-    open(joinpath(dir_paths.artifacts, "walltime_per_atmos_step.txt"), "w") do walltime_per_atmos_step_filename
-        println(walltime_per_atmos_step_filename, "$(walltime_per_atmos_step)")
+        open(joinpath(dir_paths.artifacts, "max_rss_cpu.txt"), "w") do cpu_max_rss_filename
+            cpu_max_rss_GB = Utilities.show_memory_usage()
+            println(cpu_max_rss_filename, cpu_max_rss_GB)
+        end
     end
 
-    open(joinpath(dir_paths.artifacts, "max_rss_cpu.txt"), "w") do cpu_max_rss_filename
-        cpu_max_rss_GB = Utilities.show_memory_usage()
-        println(cpu_max_rss_filename, cpu_max_rss_GB)
+    #=
+    ## Postprocessing
+    All postprocessing is performed using the root process only, if applicable.
+    Our postprocessing consists of outputting a number of plots to visualize the model output.
+
+    The postprocessing includes:
+    - Energy and water conservation checks (if running SlabPlanet with checks enabled)
+    - Animations (if not running in MPI)
+    - AMIP plots of the final state of the model
+    - Error against observations
+    - Optional additional atmosphere diagnostics plots
+    - Plots of useful coupler and component model fields for debugging
+    =#
+
+    if ClimaComms.iamroot(comms_ctx)
```

Comment:
It would be nice to be able to disable postprocessing to save time when calibrating